content/security.md

```diff
@@ -0,0 +1,380 @@
+Meteor.publish('list', function (listId) {
+  return Lists.find({
+    _id: listId,
+    userId: this.userId
+  }, {
+    fields: {
+      name: 1,
+      incompleteCount: 1,
+      userId: 1
+    }
+  });
+});
+```
+
+In the first example, if the `userId` property on the selected list changes, the query in the publication will still return the data, since the security check in the beginning will not re-run. In the second example, we have fixed this by putting the security check in the returned query itself.
+
+Unfortunately, not all publications are as simple to secure as the example above. For more tips on how to use `reywood:publish-composite` to handle reactive changes in publications, see the data loading article.
+
+#### Passing options into publications
+
+For certain applications, for example pagination, you'll want to pass options into the publication to control things like how many documents should be sent to the client. There are some extra considerations to keep in mind for this particular case.
+
+1. **Passing a limit**: In the case where you are passing the `limit` option of the query from the client, make sure to set a maximum limit. Otherwise, a malicious client could request too many documents at once, which could raise performance issues.
+2. **Passing in a filter**: If you want to pass fields to filter on because you don't want all of the data, for example in the case of a search query, make sure to intersect the fields passed from the client with the fields it is allowed to see. Otherwise, a client could query on secret fields it's not supposed to be able to access.
+3. **Passing in fields**: If you want the client to be able to decide which fields of the collection should be fetched, make sure to intersect that with the fields that client is allowed to see, so that you don't accidentally send secret data to the client.
+
+### Served files
+
+Publications are not the only place the client gets data from the server. The set of source code files and static assets that are served by your application server could also potentially contain sensitive data:
+
+1. Business logic an attacker could analyze to find weak points
+1. Secret algorithms that a competitor could steal
+1. Secret API keys
+
+While the client-side UI of your application is basically open source, every application will have some secret code on the server that you don't want to share with the world.
+
+Secret business logic in your app should be located in code that only runs on the server. This means it is in the `server/` directory of your app, in a package that is only included on the server, or in a file inside a package that was loaded only on the server.
+
+Keep in mind that code inside `if (Meteor.isServer)` blocks is still sent to the client, it is just not executed. So don't put any secret code in there.
+
+Secret API keys should never be stored in your source code at all, the next section will talk about how to handle them.
```

Comment:
Maybe say something about splitting methods up if you want to latency compensate the non-secret part -- it's a common pattern to blindly put all methods in `server/` because you are worried about a few lines of code.